learning/tour-of-beam/frontend/lib/pages/tour/widgets/unit_content.dart

```diff
@@ -17,66 +17,212 @@
  */
 
 import 'package:flutter/material.dart';
-import 'package:flutter_markdown/flutter_markdown.dart';
-import 'package:markdown/markdown.dart' as md;
 import 'package:playground_components/playground_components.dart';
 
+import '../../../constants/sizes.dart';
 import '../../../models/unit_content.dart';
+import '../state.dart';
+import 'complete_unit_button.dart';
+import 'hints.dart';
+import 'markdown/tob_markdown.dart';
+import 'solution_button.dart';
 
 class UnitContentWidget extends StatelessWidget {
+  final TourNotifier tourNotifier;
+
+  const UnitContentWidget(this.tourNotifier);
+
+  @override
+  Widget build(BuildContext context) {
+    final themeData = Theme.of(context);
+
+    return Container(
+      height: MediaQuery.of(context).size.height -
+          BeamSizes.appBarHeight -
+          TobSizes.footerHeight,
+      decoration: BoxDecoration(
+        color: themeData.backgroundColor,
+        border: Border(
+          left: BorderSide(color: themeData.dividerColor),
+        ),
+      ),
+      child: AnimatedBuilder(
+        animation: tourNotifier,
+        builder: (context, child) {
+          final currentUnitContent = tourNotifier.currentUnitContent;
+
+          return Column(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              Expanded(
+                child: _Content(
+                  tourNotifier: tourNotifier,
+                  unitContent: currentUnitContent,
+                ),
+              ),
+              _ContentFooter(tourNotifier),
+            ],
+          );
+        },
+      ),
+    );
+  }
+}
+
+class _Content extends StatelessWidget {
+  final TourNotifier tourNotifier;
+  final UnitContentModel? unitContent;
+
+  const _Content({
+    required this.unitContent,
+    required this.tourNotifier,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final content = unitContent;
+
+    if (content == null) {
+      return Container();
+    } else if (content.hints != null) {
```

Comment:
No `else` after return.

Reply:
Is `hints` the source of truth to tell a challenge from an ordinary unit? I believe we should have a better flag. Do we have anything like 'unit type' coming from the backend?

Reply:
Is it always bad to use `else` after `return`? Here's a contrary opinion: https://blog.mozilla.org/nnethercote/2009/08/31/no-else-after-return-considered-harmful/